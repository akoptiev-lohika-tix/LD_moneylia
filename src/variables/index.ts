export * from './text-variables';
export * from './style-variables';