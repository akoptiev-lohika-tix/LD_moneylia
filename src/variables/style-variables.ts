export const colors = {
  common: {
    white: '#ffffff',
  },
  primary: {
    main: '#3a3b7b',
    light: '#f8f4ff',
    dark: '#7476ed',
  },
  grey: {
    main: '#6f6d7b',
    dark: '#312e43',
    light: '#83828e',
  },

  error: {
    main: '#f83e5a',
  },
};

export const shadows = {
  shadow1: {
    shadowColor: 'rgba(0, 0, 0, 0.08)',
    shadowOffset: { width: 0, height: 0 },
  },
};
