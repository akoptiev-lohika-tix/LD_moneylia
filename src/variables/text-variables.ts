// General
export const ERROR_GENERAL_MESSAGE =
  'Oops...Your request is unavailable now. Please try again later or contact us';

// Join-screen
export const JOIN_SCREEN_TITLE = 'Save your money';
export const JOIN_SCREEN_DESCRIPTION =
  'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Est in quis erat a sit.';
export const JOIN_SCREEN_BUTTON_TEXT = 'Join for free';
export const JOIN_SCREEN_LINK_TEXT = "Don't have SPID or CIE? Find out more";

// Dashboard
export const WELCOME_TEXT = 'Welcome in Moneylia';
export const SUMMARY_TITLE = 'User summary';
export const USER_SUMMARY_STATUS_TITLE = 'Status';
export const USER_SUMMARY_BIRTHDAY_TITLE = 'Birthday';
export const USER_SUMMARY_CITY_TITLE = 'City';
export const USER_SUMMARY_COUNTRY_TITLE = 'Country';
export const USER_SUMMARY_NUMBER_TITLE = 'Number';
export const DASHBOARD_CONTRIBUTIONS_PAID_TITLE = 'Paid';
export const DASHBOARD_CONTRIBUTIONS_DUE_TITLE = 'Due';
export const DASHBOARD_CONTRIBUTIONS_ADVANCED_PAYMENT_TITLE = 'Advance payment';
export const DASHBOARD_CONTRIBUTIONS_MODULARITY_TITLE = 'Modularity';

// Not Found Screen
export const NOT_FOUND_SCREEN_TEXT =
  'Seems like this part of app is developing now. Please come back later';

// Tax Payments
export const TAX_PAYMENTS_TOP_BAR_TITLE = 'Tax payments';
export const TAX_PAYMENTS_CONTENT_TITLE = 'Movements to pay';
export const TAX_PAYMENTS_CONTENT_DESCRIPTION =
  'View pending transactions for the past two years and proceed to payment.';
