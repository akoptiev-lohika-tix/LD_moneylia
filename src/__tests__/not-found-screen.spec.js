import React from 'react';
import { render, screen, waitFor } from '@testing-library/react-native';
import renderer from 'react-test-renderer';

import { NOT_FOUND_SCREEN_TEXT } from '../variables';
import NotFoundScreen from '../screens/not-found-screen/not-found-screen';

test('NotFoundScreen renders correctly', async () => {
  const tree = await waitFor(() => renderer.create(<NotFoundScreen />).toJSON());
  expect(tree).toMatchSnapshot();
});

describe('NotFoundScreen testing', () => {
  beforeEach(() => {
    render(<NotFoundScreen />);
  });

  it('should have correct text', () => {
    expect(screen.findByText(NOT_FOUND_SCREEN_TEXT)).toBeTruthy();
  });

  it('should have icon', () => {
    expect(screen.findByTestId('not-found-icon')).toBeTruthy();
  });
});
