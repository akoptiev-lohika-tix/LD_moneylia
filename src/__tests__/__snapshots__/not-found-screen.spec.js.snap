// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundScreen renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#f8f4ff",
      "flex": 1,
      "paddingHorizontal": 20,
      "paddingVertical": 120,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#ffffff",
        "borderRadius": 12,
        "flexDirection": "row",
        "paddingHorizontal": 12,
        "paddingVertical": 48,
        "shadowColor": "rgba(0, 0, 0, 0.08)",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
      }
    }
  >
    <RNSVGSvgView
      bbHeight={28}
      bbWidth={28}
      color={4294458970}
      fill="none"
      focusable={false}
      height={28}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "flex": 0,
            "height": 28,
            "width": 28,
          },
        ]
      }
      testID="not-found-icon"
      tintColor={4294458970}
      width={28}
    >
      <RNSVGGroup
        fill={null}
        propList={
          Array [
            "fill",
          ]
        }
      >
        <RNSVGPath
          clipRule={0}
          d="M7.269 2.586A2 2 0 0 1 8.683 2h6.634a2 2 0 0 1 1.414.586l4.683 4.683A2 2 0 0 1 22 8.683v6.634a2 2 0 0 1-.586 1.414l-4.683 4.683a2 2 0 0 1-1.414.586H8.683a2 2 0 0 1-1.414-.586l-4.683-4.683A2 2 0 0 1 2 15.317V8.683a2 2 0 0 1 .586-1.414l4.683-4.683ZM15.317 4H8.683L4 8.683v6.634L8.683 20h6.634L20 15.317V8.683L15.317 4ZM10.95 15.45a1 1 0 0 1 1-1h.1a1 1 0 0 1 1 1v.1a1 1 0 0 1-1 1h-.1a1 1 0 0 1-1-1v-.1Zm2.05-7a1 1 0 1 0-2 0v4a1 1 0 0 0 2 0v-4Z"
          fill={4294458970}
          fillRule={0}
          propList={
            Array [
              "fill",
              "fillRule",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <Text
      style={
        Object {
          "color": "#3a3b7b",
          "fontFamily": "Work Sans SemiBold",
          "fontSize": 18,
          "lineHeight": 20,
          "marginLeft": 10,
        }
      }
    >
      Seems like this part of app is developing now. Please come back later
    </Text>
  </View>
</View>
`;
